python/python/lance/sampler.py

```diff
@@ -69,15 +69,15 @@ def _efficient_sample(
     chunk_sample_size = n // max_takes
     for idx, i in enumerate(range(0, total_records, chunk_size)):
         # Add more randomness within each chunk.
-        offset = i + np.random.randint(0, chunk_size - chunk_sample_size)
+        offset = min(total_records - chunk_sample_size, i + np.random.randint(0, chunk_size - chunk_sample_size))
```

Comment:
Instead of altering the outputs of sampling, why not fix the inputs? I think all you have to do is change `chunk_size` in the last iteration to be the remaining size (`total_records - i`) here.

```python
local_size = min(chunk_size, total_records - i)
```

Reply:
That would also produce a more uniform distribution in the last chunk.